### PR TITLE
Add support for including JsonTypeInfo Annotation

### PR DESCRIPTION
**Pull request by ddcruver**
Add support for including the Jackson 1 & 2 JsonTypeInfo Annotation to generated classes.

### CODE REVIEW

File: jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java [hunk 1]
```diff
@@ -156,6 +156,26 @@ public JType apply(String nodeName, JsonNode node, JsonNode parent, JPackage _pa
...
+    private void addJsonTypeInfoIfConfigured(JDefinedClass jclass, JsonNode node)
+    {
+        // Process JsonTypeInfo configuration from generation config
+        if(ruleFactory.getGenerationConfig().isIncludeJsonTypeInfoAnnotation()) {
+            // Have per-schema JavaTypeInfo configuration override what is defined in generation config; backward comparability
+            if (node.has("deserializationClassProperty")) {
+                String annotationName = node.get("deserializationClassProperty").asText();
+                addJsonTypeInfoAnnotation(jclass, annotationName);
+            } else {
+                addJsonTypeInfoAnnotation(jclass, "@class");
```
[the remark this answers is not in the document]
Comment (ddcruver) [reply jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java]: Done.

File: jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java [hunk 2]
```diff
@@ -246,13 +266,33 @@ private JDefinedClass createClass(String nodeName, JsonNode node, JPackage _pack
 
     }
 
-    private void addJsonTypeInfoAnnotation(JDefinedClass jclass, JsonNode node) {
-        if (ruleFactory.getGenerationConfig().getAnnotationStyle() == AnnotationStyle.JACKSON2) {
-            String annotationName = node.get("deserializationClassProperty").asText();
+    private void addJsonTypeInfoAnnotation(JDefinedClass jclass, String propertyName) {
```
[the remark this answers is not in the document]
Comment (ddcruver) [reply jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ObjectRule.java]: Done.

**Comment by ddcruver** [comment]
I think I am done with all my changes.  I additionally made some additional changes but nothing big.

**Comment by ddcruver** [comment]
@joelittlejohn I believe I have addressed all your review comments.

**Comment by joelittlejohn** [comment]
@ddcruver Amazing. Thank you! :+1:

**Pull request merged**